src/main/java/by/custom/utilcalculator/controller/MessageRouter.java

```diff
@@ -36,22 +36,24 @@ public SendMessage route(Update update) {
     private SendMessage getCheckInputMessageAndGetAnswer(Message message) {
         String usersMessage = message.getText();
         String userID = message.getChatId().toString();
-        UserProgress userProgress = UserStorageManager.getInstance().getUserProgress(userID);
         String answer;
 
         switch (usersMessage) {
-            case Command.START -> answer = getGreetingMessage();
+            case Command.START -> {
+                userProgressManager.createNewUserProgress(message);
+                answer = getGreetingMessage();
+            }
             case Command.EAES, Command.OTHER_COUNTRIES ->
-                    answer = botFieldsManager.processCarOrigin(usersMessage, userProgress);
+                    answer = userProgressManager.processCarOrigin(usersMessage, message);
```

Comment:
just userId should be enough.
If it's possible it's better that `userProgressManager` don't know about telegram api.
```suggestion
                    answer = userProgressManager.processCarOrigin(userId, usersMessage);
```

Reply:
Fixed.